pytest.ini

```diff
@@ -1,6 +1,17 @@
 [pytest]
 addopts=--doctest-modules --doctest-glob=pkg_resources/api_tests.txt -r sxX
-norecursedirs=dist build *.egg setuptools/extern pkg_resources/extern pkg_resources/tests/data tools .* setuptools/_vendor pkg_resources/_vendor
+norecursedirs =
+    dist
+    docs
```

Comment:
Apparently, pytest attempts to import Python modules from the docs dir too. I'm pretty sure this shouldn't be happening.

Reply:
The `--doctest-modules` means that by default pytest will discover any and all Python modules and test the docs strings in them. This approach has the advantage that all Python modules get at least syntax checking. This behavior is employed across the projects I maintain and it helps catch errors early. It's a feature, not a bug. Please revert.

Reply:
But the only Python module there is `conf.py` and it doesn't have any doctests inside. This was necessary because I used f-stings and it makes pytest fail altogether in all old envs.
Do you really want conf.py to be compatible with Python 2.7?

If you're concerned about the syntax checking, it wouldn't get worse since Travis CI always runs Sphinx and Netlify integration runs per PR too. (Did you know you can now enable RTD integration with PRs too?)

Everything that executes Sphinx config seems to be running Python 3.8. So do you still want this reverted? I don't see any compatibility guarantees for docs documented...

Reply:
I just saw https://github.com/pypa/setuptools/commit/896ef66ff3b167a13a5efb128013d5bd53633a79. Would it work if I put `'docs/conf.py'` to `collect_ignore` instead?

Reply:
P.S. This test collection error is not triggered by `--doctest-modules` because it only matches `*.txt` files.

Reply:
> Do you really want conf.py to be compatible with Python 2.7?

Python 2 support has been dropped in master, so Python 3.5 is the minimum. I'd be okay with skipping collection of docs/* on Python 3.5, but I don't want to skip collection of it on the latest stable Python (or whatever Pythons are required to build docs).

Probably we're better off just to push on dropping support for Python 3.5.

> Did you know you can now enable RTD integration with PRs too?

I did not. I'd be open to replacing or supplementing the Netlify integration with RTD previews.

> Everything that executes Sphinx config seems to be running Python 3.8. So do you still want this reverted?

Allowing `--doctest-modules` on `docs/conf.py` also means that file gets linted with flake8 and other plugins applied. So there is value in including it in the test suite, and I'd want a compelling reason to deviate from that approach for this one project.

> Would it work if I put `'docs/conf.py'` to `collect_ignore` instead?

Yes, conditional on Python 3.5, or just use Python 3.5-compatible syntax until support is dropped. Thanks.

> This test collection error is not triggered by `--doctest-modules` because it only matches `*.txt` files.

Not according to my experience or the examples in [the docs](https://docs.pytest.org/en/stable/doctest.html#doctest-integration-for-modules-and-test-files). "modules" means Python modules in addition to any files matching doctest-glob.

Reply:
> I'd be open to replacing or supplementing the Netlify integration with RTD previews.

You can see an example of the `Build pull requests for this project` toggle enabled @ https://readthedocs.org/dashboard/cheroot/advanced/ + docs are here https://docs.readthedocs.io/en/latest/guides/autobuild-docs-for-pull-requests.html. It's been public for a few months but I've been using it earlier in limited beta and it works fine unless you specify extra env vars in their UI (then they are unavailable).

> Not according to my experience or the examples in the docs. "modules" means Python modules in addition to any files matching doctest-glob.

I said `*.txt` because the docs say that the default is `test*.txt` and `pytest.ini` extends that with `--doctest-glob=pkg_resources/api_tests.txt` that adds one more file. Ans so it's safe to say that only `.txt` files will be evaluated by doctest. But what I originally meant is that the built-in test collection mechanism hits `conf.py` w/o any interactions with doctest.

> also means that file gets linted with flake8 and other plugins applied

Ah, I had no idea that flake8 is integrated via pytest... I'll make all the necessary changes.